game-core/src/main/java/games/strategy/triplea/delegate/Matches.java

```diff
@@ -106,12 +106,36 @@ private Matches() {}
     return unit -> UnitAttachment.get(unit.getType()).getIsSea();
   }
 
-  public static Predicate<Unit> unitIsSub() {
-    return unit -> UnitAttachment.get(unit.getType()).getIsSub();
+  // public static Predicate<Unit> unitIsSub() {
+  // return unit -> UnitAttachment.get(unit.getType()).getIsSub();
+  // }
+
+  public static Predicate<Unit> unitHasSubBattleAbilities() {
+    return unitCanEvade().or(unitIsFirstStrike()).or(unitCantBeTargetedByAll());
+  }
+
+  public static Predicate<Unit> unitCanEvade() {
+    return unit -> UnitAttachment.get(unit.getType()).getCanEvade();
+  }
+
+  public static Predicate<Unit> unitIsFirstStrike() {
+    return unit -> UnitAttachment.get(unit.getType()).getIsFirstStrike();
+  }
+
+  public static Predicate<Unit> unitCanMoveThroughEnemies() {
+    return unit -> UnitAttachment.get(unit.getType()).getCanMoveThroughEnemies();
+  }
+
+  public static Predicate<Unit> unitCanBeMovedThroughByEnemies() {
+    return unit -> UnitAttachment.get(unit.getType()).getCanBeMovedThroughByEnemies();
+  }
+
+  public static Predicate<Unit> unitCantTargetAll() {
+    return unit -> !UnitAttachment.get(unit.getType()).getCantBeTargetedBy().isEmpty();
```

Comment:
Should this be?:

    return unit -> !UnitAttachment.get(unit.getType()).getCantTarget().isEmpty();

Reply:
Nice catch. You are correct. Its fixed now.